open_spiel/python/examples/leduc_nfsp.py

```diff
@@ -94,66 +109,259 @@ def action_probabilities(self, state, player_id=None):
 
     self._obs["current_player"] = cur_player
     self._obs["info_state"][cur_player] = (
-        state.information_state_tensor(cur_player))
+      state.information_state_tensor(cur_player))
     self._obs["legal_actions"][cur_player] = legal_actions
 
     info_state = rl_environment.TimeStep(
-        observations=self._obs, rewards=None, discounts=None, step_type=None)
+      observations=self._obs, rewards=None, discounts=None, step_type=None)
 
     with self._policies[cur_player].temp_mode_as(self._mode):
       p = self._policies[cur_player].step(info_state, is_evaluation=True).probs
     prob_dict = {action: p[action] for action in legal_actions}
     return prob_dict
 
 
+def latest_checkpoint_dir():
+  return os.path.join(FLAGS.checkpoint_dir, "latest")
+
+
+def checkpoint_dir(episode):
+  return os.path.join(FLAGS.checkpoint_dir, str(episode))
+
+
+def list_saved_checkpoints():
+  return list(sorted(int(p) for p in os.listdir(FLAGS.checkpoint_dir) if p != "latest"))
+
+
+def most_recent_checkpoints(n):
+  return list_saved_checkpoints()[-n + 1:-1]
+
+
+def random_checkpoints(n):
+  checkpoints = list_saved_checkpoints()[:-1]
+  if n > len(checkpoints):
+    n = len(checkpoints)
+  return random.sample(checkpoints, n)
+
+
+def stratified_checkpoints(n):
+  checkpoints = list_saved_checkpoints()[:-1]
+  if n > len(checkpoints):
+    n = len(checkpoints)
+  if n == 0:
+    return []
+  elif n == 1:
+    return [checkpoints[0]]
+  else:
+    return [checkpoints[int(i * (len(checkpoints) - 1) / (n - 1))] for i in range(n)]
+
+
+class OpponentDistribution(metaclass=abc.ABCMeta):
```

Comment:
Can you move these helper classes and the two helper functions to their own file, maybe something like eval_utils.py python/algorithms? For two reasons: they seem to be written in a general way, and they reduce the complexity of the example. Also can you document the file to explain roughly how these classes are supposed to be used?